hepcrawl/spiders/alpha_spider.py

```diff
@@ -0,0 +1,150 @@
+            "/p[contains(text(),'Thesis')]/text()"
+        ).extract()
+        author_list = re.sub(r'[\n\t\xa0]', '', author_line[0]).split(
+            ",")  # Author name might contain unwanted characters.
+        author = author_list[0]
+        surname, given_names = split_fullname(author, surname="last")
+
+        for i in author_list:
+            if "thesis" in i.lower():
+                thesis_type = i.strip()
+            if "university" in i.lower():
+                affiliation = re.sub(r"[^A-Za-z\s]+", '', i).strip()
+            if self.has_numbers(i):
+                # Affiliation element might include the year
+                year = re.findall(r'\d+', i)[0].strip()
+
+        authors = []
+        authors.append({
+            # 'fullname': surname + ", " + given_names,
+            'surname': surname,
+            'given_names': given_names,
+            'affiliations': [{"value": affiliation}]
+        })
+
+        return authors, thesis_type, year
+
+    def get_abstract(self, thesis):
+        """Returns a unified abstract, if divided to multiple paragraphs.
+        """
+        abs_paragraphs = thesis.xpath(
+            "./div[@class = 'content clearfix']//div[@class='field-item even']"
+            "/p[normalize-space()][string-length(text()) > 0][position() < last()]/text()"
+        ).extract()
+        whole_abstract = " ".join(abs_paragraphs)
+        return whole_abstract
+
+    def get_title(self, node):
+        title = node.xpath(
+            "./div[@class = 'node-headline clearfix']//a/text()").extract()
+        rel_url = node.xpath(
+            "./div[@class = 'node-headline clearfix']//a/@href").extract()
+        urls = [urljoin(self.domain, rel_url[0])]
+        return title, urls
+
+    def parse(self, response):
+        """Parse Alpha web page into a HEP record."""
+
+        # Random <br>'s will create problems
+        response = response.replace(body=response.body.replace('<br />', ''))
+        node = response.selector
+
+        for thesis in node.xpath(self.itertag):
+            record = HEPLoader(
+                item=HEPRecord(), selector=thesis, response=response)
+
+            authors, thesis_type, year = self.parse_author_data(thesis)
+
+            record.add_value('authors', authors)
+            record.add_value('date_published', year)
+            record.add_value('thesis', {'degree_type': thesis_type})
+
+            title, urls = self.get_title(thesis)
+            record.add_value('title', title)
+            record.add_value('urls', urls)
+
+            abstract = self.get_abstract(thesis)
+            record.add_value("abstract", abstract)
+
+            record.add_xpath(
+                'files', "./div[@class = 'content clearfix']//span[@class='file']/a/@href")
+            record.add_value('source', 'Alpha experiment')
+            record.add_value('collections', ['HEP', 'THESIS'])
+
+            if "phd" in thesis_type.lower():
```

Comment:
To save wasted computation if the `thesis_type` is not `"phd"`, you can put the check further up and change it to use `continue`. E.g.

```python
for thesis in node.xpath(self.itertag):
    ....
    authors, thesis_type, year = self.parse_author_data(thesis)
    if "phd" not in thesis_type.lower():
        continue  # this record will be ignored
    ....
```